service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DiaryPlanStatementMapper.java

```diff
@@ -0,0 +1,147 @@
+            .isNested(isNested)
+            .id(messageContext.getIdMapper().getOrNew(ResourceType.ProcedureRequest, procedureRequest.getId()))
+            .availabilityTime(buildAvailabilityTime(procedureRequest));
+
+        buildEffectiveTime(procedureRequest).map(builder::effectiveTime);
+        buildText(procedureRequest).map(builder::text);
+
+        return TemplateUtils.fillTemplate(PLAN_STATEMENT_TEMPLATE, builder.build());
+    }
+
+    private Optional<String> buildEffectiveTime(ProcedureRequest procedureRequest) {
+        Date date = null;
+        if (procedureRequest.hasOccurrenceDateTimeType()) {
+            date = procedureRequest.getOccurrenceDateTimeType().getValue();
+        } else if (procedureRequest.hasOccurrencePeriod()) {
+            Period occurrencePeriod = procedureRequest.getOccurrencePeriod();
+            date = occurrencePeriod.hasEnd() ? occurrencePeriod.getEnd() : occurrencePeriod.getStart();
+        }
+
+        return Optional.of(formatEffectiveDate(date));
+    }
+
+    private String formatEffectiveDate(Date date) {
+        return date != null ? String.format(FULL_DATE, DateFormatUtil.formatDate(date)) : EMPTY_DATE;
+    }
+
+    private String buildAvailabilityTime(ProcedureRequest procedureRequest) {
+        if (procedureRequest.hasAuthoredOn()) {
+            return DateFormatUtil.formatDate(procedureRequest.getAuthoredOn());
+        }
+
+        throw new EhrMapperException(
+            String.format("ProcedureRequest: %s does not have required authoredOn", procedureRequest.getId())
+        );
+    }
+
+    private Optional<String> buildText(ProcedureRequest procedureRequest) {
+        return Stream.of(
+            getRequester(procedureRequest),
+            getReasonCode(procedureRequest),
+            getNotes(procedureRequest)
+        )
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .reduce(joining());
+    }
+
+    private Optional<String> getReasonCode(ProcedureRequest procedureRequest) {
+        return procedureRequest.getReasonCode()
+            .stream()
+            .map(reasonCode -> {
+                if (StringUtils.isNoneBlank(reasonCode.getText())) {
+                    return reasonCode.getText();
+                }
+                return getDisplay(reasonCode);
+            })
+            .filter(StringUtils::isNoneBlank)
+            .reduce(joining());
+    }
+
+    private String getDisplay(org.hl7.fhir.dstu3.model.CodeableConcept reasonCode) {
+        return reasonCode.getCoding()
+            .stream()
+            .map(Coding::getDisplay)
+            .reduce(joining()).orElse(StringUtils.EMPTY);
+    }
+
+    private Optional<String> getNotes(ProcedureRequest procedureRequest) {
+        return procedureRequest.getNote()
+            .stream()
+            .map(Annotation::getText)
+            .reduce(joining());
+    }
+
+    private BinaryOperator<String> joining() {
```

Comment:
Confirm with @dmcavenue1 how to join fields. I think they should be joined with spaces not commas.

Reply:
Yes, it should be spaces